clients/include/rocblas_datatype2string.hpp

```diff
@@ -228,4 +231,83 @@ inline rocblas_datatype string2rocblas_datatype(const std::string& value)
     // clang-format on
 }
 
+template <typename T, typename std::enable_if<!is_complex<T>>::type* = nullptr>
+inline T string2rocblas_datavalue(const std::string& value)
+{
+    if(value == "")
+        return 0;
+
+    if(value == "NaN")
+        return static_cast<T>(rocblas_nan_rng());
+
+    std::string::size_type sz;
+    T                      res = std::stod(value, &sz);
+
+    if(sz != value.size())
+        throw std::invalid_argument("Invalid Complex Value");
+
+    return res;
+}
+
+template <typename T, typename std::enable_if<is_complex<T>>::type* = nullptr>
+inline T string2rocblas_datavalue(const std::string& value)
+{
+    std::string::size_type sz;
+    T                      res{};
+
+    if(value == "")
+        return res;
+
+    if(value == "NaN")
+        return static_cast<T>(rocblas_nan_rng());
+
+    decltype(res.x) tmp = std::stod(value, &sz);
+
+    if(value[sz] != 'i')
+    {
+        // real part
+        res.x = tmp;
+
+        // imagnary part
```

Comment:
typo: imaginary

Reply:
fix typo in commit (fix typo "imaginary")